.github/workflows/release.yml

```diff
@@ -66,15 +75,29 @@ jobs:
       - name: Setup PostgreSQL
         uses: ikalnytskyi/action-setup-postgres@v6
 
-      - uses: actions-rs/cargo@v1
-        with:
-          command: build
-          args: --release
+      - run: |
+          if [ -z "${CROSS_COMPILE_TARGET}" ]; then
+            # normal build targetting the CPU architecture and the OS of the host
+            cargo build --release
+          else
+            # cross build with the specified rustc target
+            cargo install cross
+            cross build --release --target ${CROSS_COMPILE_TARGET}
+          fi
+        env:
+          CROSS_COMPILE_TARGET: ${{ matrix.os.cross_compile_target }}
 
       - id: build
         run: |
-          rustc --print cfg | grep = > rustc.vars
-          source rustc.vars
+          if [ -z "${CROSS_COMPILE_TARGET}" ]; then
+            # normal build targetting the CPU architecture and the OS of the host
+            rustc --print cfg | grep = > rustc.vars
```

Comment:
Alternatively, we can get rid of calling `rustc` to set `target_os` and `target_arch`, and specify the values for both statically in the config.